src/simulator/main.cc

```diff
@@ -1,29 +1,92 @@
 #include "simulation.hh"
 
+#include <boost/program_options.hpp>
 #include <nova/io.h>
 #include <nova/json.h>
 #include <nova/utils.h>
 
+#include <iostream>
 #include <ranges>
 #include <span>
 #include <string>
 
+namespace po = boost::program_options;
+
 using json = nova::json;
 
 
+void validate_format(const std::string& format) {
+#ifdef ROS2_BUILD
+    const auto valid_formats = std::vector<std::string>{ "xyz", "rosbag2" };
+#else
+    const auto valid_formats = std::vector<std::string>{ "xyz" };
+#endif
+
+    if (std::find(valid_formats.begin(), valid_formats.end(), format) == valid_formats.end()) {
+        throw po::validation_error(po::validation_error::invalid_option_value, "format", format);
+    }
+}
+
+auto parse_args(int argc, char* argv[]) -> std::optional<po::variables_map> {
+    auto arg_parser = po::options_description("Simulator");
+    arg_parser.add_options()
+        ("config,c", po::value<std::string>()->required()->value_name("FILE"), "Config file")
+        ("map,m", po::value<std::string>()->required()->value_name("FILE"), "Map file")
+        ("path,p", po::value<std::string>()->required()->value_name("FILE"), "Path file")
+#ifdef ROS2_BUILD
+        ("format,f", po::value<std::string>()->required()->value_name("xyz|rosbag2")->notifier(validate_format)->default_value("xyz"), "Output format")
+#else
+        ("format,f", po::value<std::string>()->required()->value_name("xyz")->notifier(validate_format)->default_value("xyz"), "Output format")
```

Comment:
Same as before. Does `--format` has a usefulness if there is only one possibility, which is the default?

Reply:
Maybe in the future it can be extended.